internal/notification/slack/slack.go

```diff
@@ -0,0 +1,188 @@
+package slack
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/saucelabs/saucectl/internal/report"
+
+	"github.com/rs/zerolog/log"
+	"github.com/slack-go/slack"
+
+	"github.com/saucelabs/saucectl/internal/config"
+	"github.com/saucelabs/saucectl/internal/credentials"
+)
+
+// Reporter represents reporter for slack
+type Reporter struct {
+	Channels       []string
+	TestResults    []report.TestResult
+	Framework      string
+	Metadata       config.Metadata
+	TestEnv        string
+	RenderedResult string
+	Config         config.Notifications
+	Service        Service
+}
+
+// Add adds the TestResult to the reporter. TestResults added this way can then be rendered out by calling Render().
+func (r *Reporter) Add(t report.TestResult) {
+	r.TestResults = append(r.TestResults, t)
```

Comment:
Just to be thread safe here, check the other reporter implementations we have with regards to mutex.

Reply:
Done